packages/php-wasm/universal/src/lib/php-request-handler.ts

```
@@ -553,7 +586,7 @@ export class PHPRequestHandler implements AsyncDisposable {
 		rewrittenRequestUrl: URL,
 		scriptPath: string
 	): Promise<PHPResponse> {
-		let spawnedPHP: SpawnedPHP | undefined = undefined;
+		let spawnedPHP: AcquiredPHP | undefined = undefined;
 		try {
 			spawnedPHP = await this.processManager!.acquirePHPInstance({
```

What happens when the request handler with a SinglePhpInstanceManager is already handling a request and it receives another request? AFAICT, there is no semaphore used there so it will just hand over the single PHP instance for use again, even if the instance is already in use. Is that right?

If so, maybe we should add some sort of concurrency control to avoid that scenario or maybe even throw an error if we expect the caller (like Playground CLI) to be responsible for not over-assigning requests to each worker.

---

It will likely deadlock if the first request depends on the second (e.g. loopback). PHP has an internal semaphore inside the [runStream()](https://github.com/WordPress/wordpress-playground/blob/69e8dd51d6d3e525a9eaf69dc950ccb2af6d3a8c/packages/php-wasm/universal/src/lib/php.ts#L627) method. I'd say the load balancer should hold the request until there's an available worker. And we could also throw an error here, good idea.